front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -160,29 +155,104 @@ const useScenarioData = (
     });
   }, []);
 
-  return {
-    selectedTrainId,
-    selectedTrainSummary: selectedTrainId
-      ? trainScheduleSummariesById.get(selectedTrainId)
-      : undefined,
-    trainScheduleSummaries,
-    trainSchedules,
-    projectionData:
-      trainScheduleUsedForProjection && projectionPath
-        ? {
-            trainSchedule: trainScheduleUsedForProjection,
-            ...projectionPath,
-            projectedTrains,
-            projectionLoaderData: {
-              allTrainsProjected,
-              totalTrains: timetable.train_ids.length,
-            },
-          }
-        : undefined,
-    conflicts,
-    removeTrains,
-    upsertTrainSchedules,
-  };
+  /** Update only depature time of a train */
+  const updateTrainDepartureTime = useCallback(
+    async (trainId: number, newDeparture: Date) => {
+      const trainSchedule = trainSchedules?.find((train) => train.id === trainId);
+
+      if (!trainSchedule) {
+        throw new Error('Train non trouvé');
+      }
+
+      const trainScheduleResult = await putTrainScheduleById({
+        id: trainId,
+        trainScheduleForm: {
+          ...trainSchedule,
+          start_time: newDeparture.toISOString(),
+        },
+      }).unwrap();
+
+      setProjectedTrainsById((prev) => {
+        const newProjectedTrainsById = new Map(prev);
+        newProjectedTrainsById.set(trainScheduleResult.id, {
+          ...newProjectedTrainsById.get(trainScheduleResult.id)!,
+          departureTime: newDeparture,
+        });
+        return newProjectedTrainsById;
+      });
+
+      setTrainSchedules((prev) => {
+        const newTrainSchedulesById = {
+          ...keyBy(prev, 'id'),
+          ...keyBy([trainScheduleResult], 'id'),
+        };
+        const newTrainSchedules = sortBy(Object.values(newTrainSchedulesById), 'start_time');
+        return newTrainSchedules;
```

Comment:
To harmonize with other part l.123
```suggestion
        return sortBy(Object.values(newTrainSchedulesById), 'start_time');
```

Reply:
sure 👍🏻